### DISCUSSION TITLE
UBL - Tilt bed measures too far right

### DISCUSSION THREAD

**Discussion by bjoernbusch** [Jyers/Marlin#339] [Q&A] [answered]
When I start a print and trigger the automatic bed tilting via

G29 J ;

the measurement on the right side goes all the way to X 249 and then screw of the right wheel touches (smashes gently...) into the metal on the right. Is the following place the correct place to adjust this?

#define X_MAX_POS 245

I also have MESH_INSET set to 15, but I guess that's not being used then.

Thanks!

**Comment by marcomania75** [comment]
did you recently tried to calibrate the steps for x axis? I had the same mistake (hitting "metal-endpoint" in x and y axis) caused by setting to much "steps/mm". After measuring again and correcting the steps I have no "gently contact" anymore :-)

**Reply by bjoernbusch** [reply]
no, esteps on xyz are all on default settings.

**Reply by marcomania75** [reply]
mine are a bit lower than the default setting of 80. maybe you can check with a good visual judgment if your x axis is moving to far. Move to x0 and look how far the nozzle is away from edge of the bed. move to x235 and check how far the nozzle is moved beyond the bed. if the second distance looks "longer" your x-axis is moving to far. maybe it is measurable with a ruler. I have around 5mm left in x249-position between the screw and the metal it was smashing in. Another thing could be, that the x-endstop is to far right, but that should also be seen when you check the distance at x0 to the edge of the bed, mine is around 1.5mm away...

that were the ideas I had...

**Reply by alphatrianguli** [reply]
> mine are a bit lower than the default setting of 80. maybe you can check with a good visual judgment if your x axis is moving to far. Move to x0 and look how far the nozzle is away from edge of the bed. move to x235 and check how far the nozzle is moved beyond the bed. if the second distance looks "longer" your x-axis is moving to far. maybe it is measurable with a ruler. I have around 5mm left in x249-position between the screw and the metal it was smashing in. Another thing could be, that the x-endstop is to far right, but that should also be seen when you check the distance at x0 to the edge of the bed, mine is around 1.5mm away...
> 
> that were the ideas I had...

I checked my settings and they are fine. x235 is on the very right side but still on the bed, so the steps seem fine.

Still, the sled hits the gantry with the nut screw when doing a tilt, so the tilz size seems to be bigger than the mesh size...

**Comment by alphatrianguli** [comment]
I have the same issue, which is odd. Also on standard values on the ESTEPS of XYZ.

Is this a FW issue?

**Comment by Jyers** [comment, marked as answer]
The original PR for UBL features included a significantly increased Y max. I don't know if I ever tested with this setting as I had tuned in my bed size/positions. I'll give this a test to see if this needs some modification.

**Reply by alphatrianguli** [reply]
Because of the x offset, during the tilt, the BLTouch tries to probe in the front right corner, which it cant because the sled hits the gantry.

How can I change the size of the mesh tilt / probing area without having to compile my own firmware?

Many thanks! :)

**Reply by Jyers** [reply]
Unfortunately there is no way to modify this outside of compiling, but I should have this fixed in the next release in case you don't want to compile your own.

**Reply by alphatrianguli** [reply]
Thanks for the information and for the great FW - appreciated and donated! :)

Just out of curiosity: When will the next version be ready?

**Reply by Jyers** [reply]
Thank you so much for the support! It means so much.

The next version should be out sometime this week. There haven't been a ton of fixes since the last one, so I'm hoping to get a couple of the pull requests merged before the next one.

**Comment by BlueWings172** [comment]
@nielsrahn, @bjoernbusch 

Did you figure out a solution for this whether through FW change, G-code, screen settings ?

I have the same issue and whatever numbers i set for X and Y home offset in the screen, it does not make any difference and the printhead hits the x axis on **both** ends during UBL tilting. This started happening after installing a fan duct that is much wider than stock.

**Reply by BlueWings172** [reply]
I've been trying to avoid getting into firmware compiling ever since I got my printer. But as you said, there is no way around it to solve this.
I've tried playing with home offsets (what a miserable waste of time), adding G92 in starting Gcode, play with printhead dimensions in Machine Settings in Cura and read and watched everything I could find.

At the end, I decided to bite the bullet and venture into compiling firmware. Turns out it's pretty simple after watching this video by the amazing BV3D and solved my problem following the steps you mentioned in the other post. 
The dragon is sleeping now, we've been wrestling for the entire week !


https://www.youtube.com/watch?v=so0FfFVEmiw&ab_channel=BV3D%3ABryanVines
